tests/Resources/Page/ODH/Prometheus/Prometheus.robot

```diff
@@ -68,34 +68,59 @@ Alert Should Be Firing     # robocop: disable:too-many-calls-in-keyword
                 ${state}=    Get From Dictionary    ${sub_rule}    state
                 ${name}=    Get From Dictionary    ${sub_rule}    name
                 ${duration}=    Get From Dictionary    ${sub_rule}    duration
-                IF    '${name}' == '${alert}'
-                    ${alert_found}=    Set Variable    True
-                    # Log    Alert "${name}" (duration=${duration}): state=${state}
+
+                ${alert_found}=    Run Keyword And Return Status
+                ...    Alerts Should Be Equal    ${alert}    ${alert-duration}    ${name}    ${duration}
+
+                IF    ${alert_found}
                     IF    '${state}' == 'firing'
                         Return From Keyword    ${TRUE}
+                    ELSE
+                        Exit For Loop
                     END
                 END
             END
         END
     END
 
     IF    ${alert_found} == True
-        Fail    msg=Alert "${alert}" was found in Prometheus but it wasn't firing
+        # Log To Console    msg=Alert "${alert} ${alert-duration}" was found in Prometheus but state != firing
+        Fail    msg=Alert "${alert} ${alert-duration}" was found in Prometheus but state != firing
     ELSE
-        Fail    msg=Alert "${alert}" was not found in Prometheus firing rules
+        Log To Console    msg=ERROR: Alert "${alert} ${alert-duration}" was not found in Prometheus firing rules
+        Fail    msg=Alert "${alert} ${alert-duration}" was not found in Prometheus firing rules
+    END
+
+Alerts Should Be Equal
+    [Documentation]    Compares two alerts names and fails if they are different.
+    ...    If ${alert1-duration} is not empty, compare it also with ${alert2-duration}
+    [Tags]    Private
+    [Arguments]    ${alert1-name}    ${alert1-duration}    ${alert2-name}    ${alert2-duration}
+    Should Be Equal    ${alert1-name}    ${alert2-name}
+    IF    "${alert1-duration}" != "${EMPTY}"
```

Comment:
what if alert1 duration is Empty? Doesn't matter if alert 2 duration is not empty in that case?

Reply:
Yes, it doesn't mater in that case. This is used for DeadManSnitch and other alerts, that don't have duration defined (or is = 0)